### PR TITLE
Account financial test data

### PR DESCRIPTION
**Pull request by KevinBro**
https://trello.com/c/ppoB8khk/47-test-data-module-convert-move-lines-and-periods-from-joel-s-dumps-into-a-data-module

### CODE REVIEW

File: account_financial_test_data/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Account financial test data',
```
Comment (eLBati) [inline account_financial_test_data/__openerp__.py:6]: Please add README or description

File: account_financial_test_data/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    'author': "Taktik, Odoo Community Association (OCA)",
+    'category': 'Other',
+    'depends': [
+        "base",
+        "account",
+        "account_accountant",
+        "l10n_fr",
+    ],
+    'demo': [],
+    'data': [
```
Comment (lepistone) [inline account_financial_test_data/__openerp__.py:16]: maybe we could put all the XML under `demo`. What do you think?

File: account_financial_test_data/__openerp__.py (hunk 1 above)
Comment (lepistone) [inline account_financial_test_data/__openerp__.py:6]: agreed. what is important in the README is to say what is the purpose here: some demo data to test corner cases in the reports. You might also mention as you showed me that there is data from 2015, 2016 etc. Thanks!

**Comment by lepistone** [comment]
Needs a readme, otherwise great, thanks! 👍

**Comment by lepistone** [comment]
the build has some minor issues, please give a look @KevinBro thanks!

**Comment by tafaRU** [comment]
@KevinBro, thanks for the module.
I see travis build is red, I'll make asap a PR to fix its errors, please wait for it.

**Comment by tafaRU** [comment]
@KevinBro, https://github.com/KevinBro/account-financial-reporting/pull/1 still WIP

**Comment by tafaRU** [comment]
@KevinBro, https://github.com/KevinBro/account-financial-reporting/pull/1 is ready, please have a look at it.

File: account_financial_test_data/__openerp__.py (hunk 2 above)
Comment (KevinBro) [inline account_financial_test_data/__openerp__.py:16]: It depends.
If i put the xml data in "demo", it means all others demo data will be added.
Maybe for clarity, it is more flexible to let it as it is.
It will let all the users the choice.
But if it make no sense, I m ready to change it.

**Comment by sbidoul** [comment]
@KevinBro thanks for this work!

I think the data must go in the demo section, so other modules can depend on it to create tests, while not installing the data when installing the module in non-demo mode.

Is it possible to do it without python code (ie a pure data module)?

Also an important part of the data seems to be missing, ie the move lines.

**Comment by tafaRU** [comment]
@KevinBro, https://github.com/KevinBro/account-financial-reporting/pull/2 fixes https://travis-ci.org/OCA/account-financial-reporting/jobs/126934321

File: account_financial_test_data/models/account_installer.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import model_nameimport logging
+from openerp.osv import orm
+from openerp import models, api
+
+_logger = logging.getLogger(__name__)
+
+
+class WizardMultiChartsAccounts(orm.TransientModel):
```
Comment (tafaRU) [inline account_financial_test_data/models/account_installer.py]: Please use new API.

File: account_financial_test_data/models/account_installer.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    """
+    Execute wizard automatically without showing the wizard popup window
+    """
+    _inherit = 'wizard.multi.charts.accounts'
+
+    def auto_execute(self, cr, uid, ids=False, context=None):
+        if not context:
+            context = {}
+        context['lang'] = 'en_US'
```
Comment (tafaRU) [inline account_financial_test_data/models/account_installer.py]: What's the reason for this?

**Comment by lepistone** [comment]
Thanks again @KevinBro !

I think this is very useful even if a couple things are missing (i.e. the move lines).

I'd love to merge it with the minimum changes needed to be able to do it. I think those are:
- merge KevinBro#2 to fix a syntax error
- make travis green (maybe the above if enough)
- adding a quick checklist somewhere visible like the README with the points that are TODO (i.e. move lines).
- also, I think the readme should quickly mention what happens on installation (especially the automatic account chart wizard, which is always confusing).

Let's do this and merge! Please ping if you need help.

**Comment by lepistone** [comment]
Hi @KevinBro ! The last build complains because you removed the `auto_execute` method but it is still referenced from `data/account_auto_installer.xml`. Maybe it should be removed there also?

Thanks!

**Comment by jgrandguillaume** [comment]
Hi @KevinBro 

Would you agree to fix this so we can merge your work ? It'll be a pitty that we don't make the last 100 meters on this one..

Thanks in advance for your time,

Regards,

Joël

**Comment by vrenaville** [comment]
@KevinBro 

Thanks for this PR !!!
Could you please have a look at Travis failed job ?
it will be great to have this module into the OCA

Thanks for your help

Vincent
